### PR TITLE
Use polling when syncing repos

### LINKED ISSUE #56: Nginx timeout when syncing big repos
Syncing a big repository e.g., [ulx3s](https://github.com/emard/ulx3s) takes a while which makes Nginx returns `504`. Currently, this is considered a failure, and an error message gets displayed as if syncing failed.

### PR DESCRIPTION
**Pull request by AbdulrhmnGhanem**
closes #56

### CODE REVIEW

File: nginx/kitspace.template [hunk 1]
```diff
@@ -45,6 +45,33 @@ server {
...
+			add_header Content-Type text/plain;
+			add_header 'Access-Control-Allow-Origin' '@KITSPACE_URL';
+			add_header 'Access-Control-Allow-Methods' 'POST';
+			add_header 'Access-Control-Allow-Headers' 'cache-control, pragma, user-agent, content-type';
+			add_header 'Access-Control-Allow-Credentials' 'true' always;
+			return 200;
+		}
+		proxy_pass http://gitea:3000;
+	}
+
```
Comment (kasbah) [inline nginx/kitspace.template:74]: Why not just increas the timeout for all gitea requests?

Comment (AbdulrhmnGhanem) [reply nginx/kitspace.template:74]: If any other request took that long to resolve this means there's something wrong. The official docs advise not to use more than 75s, the default is 60s. Though for `/api/v2/repos/migrate` we know it's normal to take more than that.

**Comment by AbdulrhmnGhanem** [comment]
Replaced by #62

**Comment by AbdulrhmnGhanem** [comment]
#60 should get done first, the compilation takes too long using the original dockerfile.

**Pull request merged**